src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmApiServiceImpl.java

```diff
@@ -38,9 +35,29 @@ public UpdateMetadataResponse updateMetadata(String externalRecordId, OffsetDate
             .build();
 
         return armApiClient.updateMetadata(
-            String.format("Bearer %s", armTokenResponse.getAccessToken()),
+            getArmBearerToken(),
             armUpdateMetadataRequest
-        ).getBody();
+        );
+    }
+
+    @Override
+    public InputStream downloadArmData(String externalRecordId, String externalFileId) {
+        byte[] response = armApiClient.downloadArmData(
```

Comment:
Cant we stream this straight out rather than holding it in memory?

Reply:
or is the size not going to be significant?

Reply:
good point Michael. In fact initially I defined the return type as `InputStream` but couldn't get that to work with feign client so I switched to `byte []`. But you're making a good point let me have a look